client/log_client.go

```diff
@@ -184,33 +187,10 @@ func (c *LogClient) getAndVerifyLatestRoot(ctx context.Context, trusted *types.L
 		// Tree has not been updated.
 		return &logRoot, nil
 	}
-	// Fetch a consistency proof if this isn't the first root we've seen.
-	var consistency *trillian.GetConsistencyProofResponse
-	if trusted.TreeSize > 0 {
-		// Get consistency proof.
-		consistency, err = c.client.GetConsistencyProof(ctx,
-			&trillian.GetConsistencyProofRequest{
-				LogId:          c.LogID,
-				FirstTreeSize:  int64(trusted.TreeSize),
-				SecondTreeSize: int64(logRoot.TreeSize),
-			})
-		if err != nil {
-			return nil, err
-		}
-		// If this request hit a more out-of-date server instance than the GetSLR request,
-		// there may be an empty proof and an earlier SLR.
-		var proofRoot types.LogRootV1
-		if err := proofRoot.UnmarshalBinary(consistency.GetSignedLogRoot().GetLogRoot()); err != nil {
-			return nil, err
-		}
-		if proofRoot.TreeSize < logRoot.TreeSize {
-			return nil, fmt.Errorf("response for ConsistencyProof has a smaller (%d) root than requested (%d)", proofRoot.TreeSize, logRoot.TreeSize)
-		}
-	}
 
 	// Verify root update if the tree / the latest signed log root isn't empty.
-	if logRoot.TreeSize > 0 {
-		if _, err := c.VerifyRoot(trusted, resp.GetSignedLogRoot(), consistency.GetProof().GetHashes()); err != nil {
+	if trusted.TreeSize > 0 && logRoot.TreeSize > 0 {
```

Comment:
When `trusted.TreeSize == 0`, this line will prevent it from being updated (which is bad). 
Did `VerifyRoot` fail when `trusted.TreeSize == 0`?  We might need to fix how `VerifyRoot` handles this case.  We've had a few other instances where we didn't handle the `0` edge condition correctly. 

```suggestion
	if logRoot.TreeSize > 0 {
```

Reply:
Done.